torch/_functorch/vmap.py

```diff
@@ -391,19 +398,116 @@ def vmap(
         sequences out of the box.
     """
     _check_randomness_arg(randomness)
+    if not (chunk_size is None or chunk_size > 0):
+        raise ValueError("vmap: `chunk_size` should be greater than 0.")
+
+    def _get_chunked_inputs(flat_args, flat_in_dims, batch_size, chunk_size):
+        split_idxs = (batch_size,)
+        if chunk_size is not None and chunk_size > 0:
+            n_chunks = n_chunks = batch_size // chunk_size
+            chunk_numels = [chunk_size] * n_chunks
+            # remainder chunk
+            remainder = batch_size % chunk_size
+            if remainder != 0:
+                chunk_numels.append(remainder)
+            split_idxs = tuple(itertools.accumulate(chunk_numels))
+
+        flat_args_chunks = tuple(
+            t.tensor_split(split_idxs, dim=in_dim) if in_dim is not None else [t, ] * len(split_idxs)
```

Comment:
`tensor_split` returns a list of views : https://github.com/pytorch/pytorch/blob/7330eabe3675e62d580b3c7d453b1f7f356e2c61/aten/src/ATen/native/native_functions.yaml#L1358